homeassistant/components/tessie/coordinator.py

```diff
@@ -0,0 +1,84 @@
+"""Tessie Data Coordinator."""
+from datetime import timedelta
+from http import HTTPStatus
+import logging
+from typing import Any
+
+from aiohttp import ClientResponseError
+from tessie_api import get_state
+
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator
+
+from .const import TessieStatus
+
+# This matches the update interval Tessie performs server side
+TESSIE_SYNC_INTERVAL = 10
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class TessieDataUpdateCoordinator(DataUpdateCoordinator):
```

Comment:
It's nice to add a type annotation to `DataUpdateCoordinator` as it's generic, representing the return value of `_async_update_data` (`dict[str, Any]`).